haas/api.py

```diff
@@ -505,21 +563,72 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    """Lists all networks
+
+    """
+    get_auth_backend().require_admin()
+
+    networks = db.session.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access:
+            net = {'network_id': n.network_id, 'projects': [p.label for p in n.access]}
+        else:
+            net = {'network_id': n.network_id, 'projects': None}
+        result[n.label] = net
+            
+    return json.dumps(result, sort_keys = True)
+
+@rest_call('GET', '/network/<network>/attachments')
```

Comment:
Please add a Schema{} here and in the other API calls. See the other API calls for examples of optional arguments.

After a soon-to-be-committed PR is merged, they will become mandatory.